packages/contracts/contracts/TestContracts/invariants/echidna/EchidnaBraindeadFeedTester.sol

```diff
@@ -0,0 +1,310 @@
+    }
+
+    function setLatestEth(uint80 latestRoundId, uint256 price, uint256 updateTime) public {
+        (uint80 roundId, int256 answer, , uint256 updatedAt, ) = collEthCLFeed.latestRoundData();
+
+        latestRoundId = uint80(
+            clampBetween(
+                uint256(latestRoundId),
+                uint256(roundId),
+                uint256(roundId + MAX_ROUND_ID_CHANGE)
+            )
+        );
+        // NOTE: Updated to clamp based on proper realistic prices
+        price = clampBetween(price, MIN_ETH_VALUE, MAX_ETH_VALUE);
+
+        updateTime = (
+            clampBetween(
+                updateTime,
+                _getOldestAcceptableTimestamp(),
+                _getNewestAcceptableTimestamp()
+            )
+        );
+
+        collEthCLFeed.setLatestRoundId(latestRoundId);
+        collEthCLFeed.setPrice(int256(price));
+        collEthCLFeed.setUpdateTime(updateTime);
+    }
+
+    function setPreviousEth(
+        uint80 prevRoundId,
+        uint256 prevPrice,
+        uint256 prevUpdateTime,
+        bool flag
+    ) public {
+        (uint80 roundId, int256 answer, , uint256 updatedAt, ) = collEthCLFeed.getRoundData(0);
+        prevRoundId = uint80(
+            clampBetween(
+                uint256(prevRoundId),
+                uint256(roundId),
+                uint256(roundId + MAX_ROUND_ID_CHANGE)
+            )
+        );
+        prevPrice = (clampBetween(prevPrice, MIN_ETH_VALUE, MAX_ETH_VALUE));
+        prevUpdateTime = (
+            clampBetween(
+                prevUpdateTime,
+                _getOldestAcceptableTimestamp(),
+                _getNewestAcceptableTimestamp()
+            )
+        );
+        collEthCLFeed.setPrevRoundId(prevRoundId);
+        collEthCLFeed.setPrevPrice(int256(prevPrice));
+        collEthCLFeed.setPrevUpdateTime(prevUpdateTime);
+    }
+
+    function setLatestBTC(uint80 latestRoundId, uint256 price, uint256 updateTime) public {
+        (uint80 roundId, int256 answer, , uint256 updatedAt, ) = ethBtcCLFeed.latestRoundData();
+
+        latestRoundId = uint80(
+            clampBetween(
+                uint256(latestRoundId),
+                uint256(roundId),
+                uint256(roundId + MAX_ROUND_ID_CHANGE)
+            )
+        );
+        // NOTE: Updated to clamp based on proper realistic prices
+        price = (clampBetween(price, MIN_BTC_VALUE, MAX_BTC_VALUE));
+
+        updateTime = (
+            clampBetween(
+                updateTime,
+                _getOldestAcceptableTimestamp(),
+                _getNewestAcceptableTimestamp()
+            )
+        );
+
+        ethBtcCLFeed.setLatestRoundId(latestRoundId);
+        ethBtcCLFeed.setPrice(int256(price));
+        ethBtcCLFeed.setUpdateTime(updateTime);
+    }
+
+    function setPreviousBTC(
+        uint80 prevRoundId,
+        uint256 prevPrice,
+        uint256 prevUpdateTime,
+        bool flag
```

Comment:
flag parameter seems generally used to randomly pick between `collEthCLFeed` and `ethBtcCLFeed`

it doesn't seem to be used here - vestigal or intended to be used?

where it is used I would opt for a more descriptive variable name, perhaps `chainlinkPriceFeedFlag`